packages/material-ui/src/Stack/Stack.js

```diff
@@ -72,22 +52,37 @@ export const style = ({ styleProps, theme }) => {
   if (styleProps.spacing) {
     const transformer = createUnarySpacing(theme);
 
-    const base = Object.keys(theme.breakpoints.values).reduce((acc, breakpoint) => {
-      if (styleProps.spacing[breakpoint] != null || styleProps.direction[breakpoint] != null) {
-        acc[breakpoint] = true;
-      }
-      return acc;
-    }, {});
-
-    const directionValues = resolveBreakpointValues({ values: styleProps.direction, base });
-    const spacingValues = resolveBreakpointValues({ values: styleProps.spacing, base });
+    let spacingValues = styleProps.spacing;
+    if (typeof styleProps.direction === 'object') {
+      // add breakpoint that `direction` has but `spacing` does not
+      // by using breakpoint value from large->small
+      spacingValues = Object.keys(theme.breakpoints.values).reduce((acc, breakpoint) => {
+        if (styleProps.spacing[breakpoint] != null) {
+          acc[breakpoint] = styleProps.spacing[breakpoint];
+        } else if (styleProps.direction[breakpoint] != null) {
+          // find breakpoint value of spacing
+          acc[breakpoint] = getBreakpointValue({
+            data: styleProps.spacing,
+            breakpoint,
+            breakpointKeys: theme.breakpoints.keys,
+          });
+        }
+        return acc;
+      }, {});
+    }
```

Comment:
This seems to require more CPU cycles, does it impact performance?

Reply:
Also: Don't use `reduce` here. It's not doing what it's supposed to do (working with immutable values) and working with immutable values is obviously unnecessary here.